Zend/zend_vm_def.h

```diff
@@ -2577,6 +2577,9 @@ ZEND_VM_C_LABEL(try_assign_dim_array):
 					Z_ADDREF_P(value);
 				}
 			}
+			if (UNEXPECTED(RETURN_VALUE_USED(opline))) {
+				ZVAL_COPY(EX_VAR(opline->result.var), value);
```

Comment:
Why was this not required before? `OP2_TYPE == IS_UNUSED` is for `$foo[] = 'bar';`, right? Is there some case where this didn't work?

Reply:
This code used to be outside the if-else. But because I added special handling for the else branch I had to copy that piece of code in both branches. This branch has the original, unmodified code while the else branch has the code that uses the new API.
This case is necessary in this branch because of for example this test: `Zend/tests/assign_dim_ref_free.phpt`.

Reply:
Formatting on mobile is trash, but the modified code below lives in the else-branch and no other code in this handler was removed. I'll have another look on the desktop 🙂

Reply:
Oh, I see it now :slightly_smiling_face: